build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java

```diff
@@ -397,4 +469,20 @@ public DirectoryProperty getSourceDirectory() {
     public List<RestTestTransform<?>> getTransformations() {
         return transformations;
     }
+
+    @Input
```

Comment:
Why add these explicit properties vs annotating the `Skip` transform itself as we do with other transforms?

Reply:
These are added to account for the directory and file portion of the test. If we only relied on the `Skip` annotation, then Gradle would only track on the "test name" (as in monitoring/bulk/test name). Now that I added the @Input annotation to the Skip class, the key of the maps are only needed here (as a function of the monitoring/bulk part of the full test name).

I can simplify down to only the key names if you think that would help with readability.